### PR TITLE
Mithril 420: Contact API Client

### PR DESCRIPTION
**Pull request by msingla2**

### CODE REVIEW

File: sdk/src/main/java/com/ifountain/opsgenie/client/model/beans/Contact.java [hunk 1]
```diff
@@ -5,16 +5,17 @@
...
-public class Contact extends BeanWithId {
+public class Contact extends Bean {
 
     private Method method;
     private String to;
-    private Boolean enabled;
-    private String disabledReason;
+    private Status status;
+    private String id;
+
```
Comment (prakhar-mudaiya) [inline sdk/src/main/java/com/ifountain/opsgenie/client/model/beans/Contact.java:17]: empty line

File: sdk/src/main/java/com/ifountain/opsgenie/client/model/beans/Contact.java [hunk 2]
```diff
@@ -58,20 +67,20 @@ public Contact withTo(String to) {
...
+        this.id = id;
         return this;
     }
 
     public enum Method {
         EMAIL(OpsGenieClientConstants.API.EMAIL), SMS(OpsGenieClientConstants.API.SMS),
-        VOICE(OpsGenieClientConstants.API.VOICE),
-        MOBILE_APP(OpsGenieClientConstants.API.MOBILE_APP);
+        VOICE(OpsGenieClientConstants.API.VOICE);
+       // MOBILE_APP(OpsGenieClientConstants.API.MOBILE_APP);
```
Comment (prakhar-mudaiya) [inline sdk/src/main/java/com/ifountain/opsgenie/client/model/beans/Contact.java:83]: is this not supported now?

File: sdk/src/main/java/com/ifountain/opsgenie/client/model/contact/BaseContactRequest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.ifountain.opsgenie.client.model.contact;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.ifountain.opsgenie.client.model.BaseRequest;
+import com.ifountain.opsgenie.client.model.BaseResponse;
+
+/**
+ * @author Mehmet Mustafa Demir
```
Comment (prakhar-mudaiya) [inline sdk/src/main/java/com/ifountain/opsgenie/client/model/contact/BaseContactRequest.java:8]: Can you please remove author.  or edit to your name

File: sdk/src/main/java/com/ifountain/opsgenie/client/model/contact/AddContactRequest.java [hunk 4]
```diff
@@ -1,24 +1,39 @@
...
+import java.util.Objects;
 
 /**
  * Container for the parameters to make an add contact api call.
  *
  * @author Mehmet Mustafa Demir
  * @see com.ifountain.opsgenie.client.IContactOpsGenieClient#addContact(AddContactRequest)
  */
-public class AddContactRequest extends BaseUserRequest<AddContactResponse, AddContactRequest> {
+public class AddContactRequest extends BaseContactRequest<AddContactResponse, AddContactRequest> {
```
Comment (prakhar-mudaiya) [inline sdk/src/main/java/com/ifountain/opsgenie/client/model/contact/AddContactRequest.java:17]: Did we finalise editing current request or we were thinking of create AddContectRequestV2. @trabab ?

File: sdk/src/main/java/com/ifountain/opsgenie/client/model/contact/DisableContactRequest.java [hunk 5]
```diff
@@ -6,11 +6,11 @@
  * @author Mehmet Mustafa Demir
  * @see com.ifountain.opsgenie.client.IContactOpsGenieClient#disableContact(DisableContactRequest)
```
Comment (prakhar-mudaiya) [inline sdk/src/main/java/com/ifountain/opsgenie/client/model/contact/DisableContactRequest.java:7]: Interfaces will remain same? Do we need to update them?

**Review by prakhar-mudaiya — CHANGES_REQUESTED** [review]

File: sdk/src/main/java/com/ifountain/opsgenie/client/model/beans/Contact.java (hunk 2 above)
Comment (msingla2) [reply sdk/src/main/java/com/ifountain/opsgenie/client/model/beans/Contact.java:83]: Yes, I have removed this as per mentioned in document.
Details mentioned for this API:
This parameter is the contact method of user and should be one of email, sms or voice. Please note that adding mobile is not supported from API.

File: sdk/src/main/java/com/ifountain/opsgenie/client/model/beans/Contact.java [hunk 6]
```diff
@@ -58,20 +67,19 @@ public Contact withTo(String to) {
...
-        this.disabledReason = disabledReason;
+    public Contact withId(String id) {
+        this.id = id;
         return this;
     }
 
     public enum Method {
         EMAIL(OpsGenieClientConstants.API.EMAIL), SMS(OpsGenieClientConstants.API.SMS),
-        VOICE(OpsGenieClientConstants.API.VOICE),
-        MOBILE_APP(OpsGenieClientConstants.API.MOBILE_APP);
```
Comment (trabab) [inline sdk/src/main/java/com/ifountain/opsgenie/client/model/beans/Contact.java:74]: Any reason for removing the mobile app here

File: sdk/src/main/java/com/ifountain/opsgenie/client/model/contact/AddContactRequest.java (hunk 4 above)
Comment (trabab) [reply sdk/src/main/java/com/ifountain/opsgenie/client/model/contact/AddContactRequest.java:17]: @prakhar-mudaiya , we are checking if the v1 versions are working, if so then we need to create the separate implementation for the v2 versions keeping in mind the backward compatibility of the apis.

**Review by trabab — COMMENTED** [review]
@manisha, please ensure that the changes are backward compatible if the v1 apis are still working.

**Pull request closed without merging**